repos/system_upgrade/el7toel8/actors/detectgrubconfigerror/libraries/scanner.py

```diff
@@ -0,0 +1,14 @@
+import re
+
+
+def detect_config_error(conf_file):
+    '''
+    Checks grub configuration for syntax error in GRUB_CMDLINE_LINUX value.
+
+    :return: Function returns True if error was detected, otherwise False.
+    '''
+    with open(conf_file, 'r') as f:
+        config = f.read()
+
+    pattern = 'GRUB_CMDLINE_LINUX="[^"]+"(?!(\s*$)|(\s+GRUB))'
```

Comment:
```suggestion
    pattern = r'GRUB_CMDLINE_LINUX="[^"]+"(?!(\s*$)|(\s+GRUB))'
```
Without the _r_ marking the string as regex, Python tries to escape whatever is after _\\_, however \s is not a valid escape sequence ([flake8 W605](https://lintlyci.github.io/Flake8Rules/rules/W605.html)).